### PR TITLE
including singularity legacy without singularity

### PR DESCRIPTION
**Pull request by vsoch**
Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by vsoch** [comment]
Huh, doesn't find the package.

```bash
 ---> Running in 5aa4f33b0e23
==> Error: Package 'singularitylegacy' not found.
```

**Comment by alecbcs** [comment]
I guess I assumed Spack packages were case insensitive but that appears not to be the case?

**Comment by vsoch** [comment]
I think the `-` is just missing, at least normally on the command line I'd do:

```bash
$ spack spec singularity-legacy
```
If I leave out the dash it doesn't work.

```bash
$ bin/spack spec singularitylegacy
Input spec
--------------------------------
singularitylegacy

Concretized
--------------------------------
==> Error: Package 'singularitylegacy' not found.
```
Is this a bug with the containerize command or something binoc is doing?

**Comment by alecbcs** [comment]
So Spack must do a conversion between Camel Case and Snake Case? Because the name of the package is,
```python3
class SingularityLegacy(AutotoolsPackage):
    """Singularity is a container platform focused on supporting 'Mobility of
       Compute'. The software changed the installation method from AutoTools
       to GoLang, so we have two separate package names to support that. The
       legacy package is pre-version 3.0.0
    """
```
I'll add this into our system to make that conversion.

File: spack/s/singularity-legacy/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+       legacy package is pre-version 3.0.0
+    """
+
+    homepage = "https://www.sylabs.io/singularity/"
+    url      = "https://github.com/sylabs/singularity/releases/download/2.5.2/singularity-2.5.2.tar.gz"
+    git      = "https://github.com/sylabs/singularity.git"
+
+    # Versions before 2.5.2 suffer from a serious security problem.
+    # https://nvd.nist.gov/vuln/detail/CVE-2018-12021
+    version('2.6-release', branch='vault/2.6-release', deprecated=True)
```
Comment (alecbcs) [inline spack/s/singularity-legacy/package.py:22]: Looks like this failed because the version is deprecated. Should we un-dreprecate it?

Comment (vsoch) [reply spack/s/singularity-legacy/package.py:22]: oh hmm, let's just skip this one then, I always get nervous when I see "serious security problem" :P

**Pull request closed without merging**